packages/roosterjs-editor-plugins/lib/plugins/HyperLink/HyperLink.ts

```diff
@@ -105,4 +134,38 @@ export default class HyperLink implements EditorPlugin {
             return anchor ? anchor.href : null;
         } catch {}
     }
+
+    /**
+     * Compares the normalized URL of inner text of element to its href to see if they match
+     */
+    private doesLinkDisplayMatchHref(element: HTMLAnchorElement): boolean {
+        let display = element.innerText.trim();
+        let rule = new RegExp(`^(?:https?:\\/\\/)?${display}\\/?`, 'i');
```

Comment:
display may contain some special charactor for regular expression

Reply:
I've added a line which will find all special characters in the display text and escape them so that matching works as expected.